project/src/func.cpp

```diff
@@ -319,6 +347,39 @@ google::protobuf::Any *Func::profileEvent(Database &kv_client_,
   return any;
 }
 
+// actually deal with stream request, called by func server
+google::protobuf::Any *Func::streamEvent(Database &kv_client_, google::protobuf::Any &payload, Status &status) {
+  StreamRequest request;
+  payload.UnpackTo(&request);
+
+  // check that user exists
+  if (!isInKv(kUserFollowing + request.username(), kv_client_)) {
+    status = grpc::Status(grpc::StatusCode::NOT_FOUND, "User does not exist");
+    LOG(ERROR) << status.error_code() << ": " << status.error_message()
+               << std::endl;
+    return nullptr;
+  }
+
+  // Get all warbles' ids associated with the hash tag from kv store
+  std::vector<GetReply> get_replies = kv_client_.get(request.hashtag());
+
+  StreamReply response;
+  if (request.index() != -1) {
```

Comment:
if this is the first time for the user to request for the tag, request.index() is -1, for loop is omitted, which means no warbles in the response (so any previous warble is omitted). And in the response, the index field is set to the size of vector ( value corresponding to the hashtag in the kvstore ), which means next time, it will only read from that index, and get only warbles posted after the stream request